expr/cast.go

```diff
@@ -103,7 +106,23 @@ func castToIP(zv zng.Value) (zng.Value, error) {
 }
 
 func castToDuration(zv zng.Value) (zng.Value, error) {
-	if zng.IsFloat(zv.Type.ID()) {
+	id := zv.Type.ID()
+	if zng.IsStringy(id) {
+		s := string(zv.Bytes)
+		d, err := time.ParseDuration(s)
+		var ns int64
+		if err == nil {
+			ns = int64(d)
+		} else {
+			f, ferr := strconv.ParseFloat(s, 64)
```

Comment:
```suggestion
			f, ferr := byteconv.ParseFloat(zv.Bytes, 64)
```

Reply:
Given that the logic here already creates the string for ParseDuration, I don't think this helps much, but I'll go ahead and change it.